DRIVERS/MDIS_LL/SMB2/TOOLS/SMB2_SHC_CTRL/COM/smb2_shc_ctrl.c

```diff
@@ -394,6 +427,78 @@ static void set_temperature(int32 tempC)
 	}
 }
 
+/****************************************************************************/
+/** Print status of persistent power button
+*/
+static void print_persistent_pwrbtn_status()
+{
+	struct shc_fwversion firm_version;
+        int err = SMB2SHC_GetFirm_Ver(&firm_version);
+	if (err) {
+		PrintError("***ERROR: SMB2SHC_GetFirm_Ver:", err);
+                return err;
+	}
+        if (firm_version.min_revision <= 16) {
+                printf("***ERROR: Persistent power button is not supported in shelf controller revisions below 4.17.");
+                return;
+        }
+
+	u_int8 status;
+	err = SMB2SHC_GetPersistentPowerbuttonStatus((u_int8*)&status);
+	if (err) {
+		PrintError("***ERROR: SMB2SHC_GetPersistentPowerbuttonStatus", err);
+	}
+	else{
+		printf("Persistent Power Button Status: %s\n", status == 1 ? "ON" : "OFF");
+	}
+}
+
+/****************************************************************************/
+/** Set status of persistent power button
+*/
+static void set_persistent_pwrbtn_status(u_int32 status)
+{
+	struct shc_fwversion firm_version;
+        int err = SMB2SHC_GetFirm_Ver(&firm_version);
+	if (err) {
+		PrintError("***ERROR: SMB2SHC_GetFirm_Ver:", err);
+                return;
+	}
+        if (firm_version.min_revision <= 16) {
+                printf("***ERROR: Cannot set persistent power button status in shelf controller revisions below 4.17.");
+                return;
+        }
+        printf("Setting persistent power button to: %s\n", status == 1 ? "ON" : "OFF");
+	err = SMB2SHC_SetPersistentPowerbuttonStatus(status);
+	if (err) {
+		PrintError("***ERROR: SMB2SHC_SetPersistentPowerbuttonStatus", err);
+	}
+}
+
+/****************************************************************************/
+/** Set power cycle duration in milliseconds
+*/
+static void set_power_cycle_duration(u_int32 delay)
+{
+	struct shc_fwversion firm_version;
+        int err = SMB2SHC_GetFirm_Ver(&firm_version);
+	if (err) {
+		PrintError("***ERROR: SMB2SHC_GetFirm_Ver:", err);
+                return;
+	}
+        if (firm_version.min_revision <= 16) {
```

Comment:
Missing compare of maj_revision.
Otherwise rev. 5.1 would lead to "ERROR: ... revisions below 4.17."